akka-sse/src/main/scala/de/heikoseeberger/akkasse/ServerSentEventParser.scala

```diff
@@ -34,23 +34,35 @@ private object ServerSentEventParser {
 
   private val linePattern = """([^:]+): ?(.*)""".r
 
-  private def parseServerSentEvent(lines: Vector[String]): ServerSentEvent = {
-    val values = lines.foldLeft(Map.empty[String, Vector[String]].withDefault(_ => Vector.empty)) {
-      case (m, "")                                                      => m
-      case (m, linePattern(field @ (Data | Event | Id | Retry), value)) => m.updated(field, m(field) :+ value)
-      case (m, field)                                                   => m.updated(field, m(field) :+ "")
+  private def parseFields(lines: Vector[String]) = {
+    var data = Vector.empty[String]
+    var eventType = null: String
+    var id = null: String
+    var retry = null: String
+    for (line <- lines) {
+      line match {
+        case linePattern(field @ (Data | Event | Id | Retry), value) => field match {
+          case Data  => data :+= value
+          case Event => eventType = value
+          case Id    => id = value
+          case Retry => retry = value
+          case _     =>
+        }
+        case Data  => data :+= ""
+        case Event => eventType = ""
+        case Id    => id = ""
+        case Retry => retry = ""
+        case _     =>
+      }
     }
-    def toInt(s: String) = try Some(s.trim.toInt) catch { case NonFatal(_) => None }
-    ServerSentEvent(
-      data = values(Data).mkString(LF),
-      eventType = values(Event).lastOption,
-      id = values(Id).lastOption,
-      retry = values(Retry).lastOption.flatMap(toInt)
-    )
+    (data, eventType, id, retry)
   }
+
+  private def silentlyToInt(s: String) = try Some(s.trim.toInt) catch { case NonFatal(_) => None }
```

Comment:
I need an `Option` here, because I use this method in a `flatMap` call on another `Option`, which means other code would need to be significantly rearranged. As "retry" is rarely used, I rather keep it this way.